furusiya/game.py

```diff
@@ -1,23 +1,10 @@
-class GameMetaClass(type):
-    def __new__(mcs, class_name, bases, dct):
-        cls = type.__new__(mcs, class_name, bases, dct)
-        cls._instance = cls()
-        return cls
-
-    def __getattr__(cls, item):
-        return cls._instance.__dict__.get(item)
-
-    def __setattr__(cls, key, value):
-        if key == '_instance':
-            super().__setattr__(key, value)
-        cls._instance.__setattr__(key, value)
-
-
-class Game(metaclass=GameMetaClass):
-    _instance = None
+class Game:
+    instance = None
     _dont_pickle = {'ui', 'save_manager', 'keybinder', 'renderer'}
 
     def __init__(self):
+        Game.instance = self
```

Comment:
Wouldn't this be overwritten every time a new instance is created? If so, is it desirable?

Reply:
Yes, it will. `Game` is now both a singleton and not so.

Singletons rightly have a bad reputation in testing. Anything you touch in one test has to be "reset" or cleared in the next run, or it might accidentally affect another test. Worse, the order of test runs could affect the state of the object (if it's not cleared).

So, using a non-singleton approach makes more sense. But in the production (non-test) code, having a single instance is desirable. So this approach is a mix: you can use it as a singleton-like, or you can use multiple instances for clean testing.